src/model/operation/convolution.h

```diff
@@ -0,0 +1,93 @@
+#ifndef SINGA_MODEL_OPERATION_CONVOLUTION_H_
+#define SINGA_MODEL_OPERATION_CONVOLUTION_H_
+
+#include <string>
+#include <vector>
+#include "singa/core/tensor.h"
+#include "singa/utils/logging.h"
+
+#ifdef USE_CUDNN
+#include <cudnn.h>
+#include "../layer/cudnn_utils.h"
+#endif // USE_CUDNN
+
+
+namespace singa {
+
+class ConvHandle {
+
+ public:
+  ConvHandle(const Tensor &input, const std::vector<size_t>& kernel_size,
+             const std::vector<size_t>& stride, const std::vector<size_t>& padding,
+             const size_t in_channels, const size_t out_channels,
+             const bool bias);
+ 
+  size_t kernel_w;
```

Comment:
when the pysinga creates the convhandle, it is using int (not size_t), did you encounter any error like?
```
Traceback (most recent call last):
  File "mnist_cnn.py", line 130, in <module>
    loss, y = forward(inputs, targets)
  File "mnist_cnn.py", line 113, in forward
    y = conv1(x)
  File "/home/wangwei/miniconda3/lib/python3.6/site-packages/singa/autograd.py", line 745, in __call__
    self.inner_params['workspace_MB_limit'] * 1024 * 1024, self.inner_params['cudnn_prefer'])
  File "/home/wangwei/miniconda3/lib/python3.6/site-packages/singa/singa_wrap.py", line 2043, in __init__
    this = _singa_wrap.new_CudnnConvHandle(*args)
NotImplementedError: Wrong number or type of arguments for overloaded function 'new_CudnnConvHandle'.
  Possible C/C++ prototypes are:
    singa::CudnnConvHandle::CudnnConvHandle(singa::Tensor const &,std::vector< size_t,std::allocator< size_t > > const &,std::vector< size_t,std::allocator< size_t > > const &,std::vector< size_t,std::allocator< size_t > > const &,size_t const,size_t const,bool const,size_t const,std::string const &)
    singa::CudnnConvHandle::CudnnConvHandle(singa::Tensor const &,std::vector< size_t,std::allocator< size_t > > const &,std::vector< size_t,std::allocator< size_t > > const &,std::vector< size_t,std::allocator< size_t > > const &,size_t const,size_t const,bool const,size_t const)
    singa::CudnnConvHandle::CudnnConvHandle(singa::Tensor const &,std::vector< size_t,std::allocator< size_t > > const &,std::vector< size_t,std::allocator< size_t > > const &,std::vector< size_t,std::allocator< size_t > > const &,size_t const,size_t const,bool const)

```

Reply:
i didn't encounter this error. i use python 2.7 to do the test. maybe because the difference of python version?

Reply:
can you try python3?
```
cmake -USE_PYTHON3=ON -USE_CUDA=ON ..
```

Reply:
ok. i will try and report to you.